flux-job: add special exit code to flux job wait when not waitable

chu11 opened the pull request:
As discussed in #5033, it is hard to differentiate between a job not being waitable / no more jobs are waitable vs a waitable job had an error.  Because in both circumstances `flux job wait` exits with 1.

This PR has `flux job wait` exit with 100 when either a jobid is non-waitable (invalid, not waitable, etc.) or there are no more waitable jobs.  So we can more easily write a script like this:

```
while jobid=`flux job wait 2> /dev/null`; ec=$?; test $ec -ne 100
do
    echo "$jobid finished"
    if [ $ec -eq 0 ]
    then
        echo "job successful"
        # do some post processing on successful jobs
    else
        echo "job failed"
        # do some post processing on failed jobs
    fi
done
echo "no more jobs to wait on"
```

I picked the exit code 100 somewhat arbitrarily.  It's common to use exit code of 2, 3, 4, ... for other purposes.  I just wanted to make it clear this wasn't an exit code from the job.  Maybe 100 is overkill.  I could make it configurable??

chu11 commented:
re-pushed, doing exit code 2 instead of 100.

garlick commented on doc/man1/flux-job.rst:
```diff
@@ -119,14 +119,15 @@ results are consumed.
-a code of one.  It is an error if the job was not submitted with the
-``waitable`` flag.
+a code of one.  If the jobid is invalid or the job is not waitable, ``flux job wait``
+exits with two.  This special exit code of two is used to differentiate between
```
To be consistent, should be "a code of two".   I agree that's starting to get a little long winded but consistentcy...

garlick commented on doc/man1/flux-job.rst:
```diff
@@ -119,14 +119,15 @@ results are consumed.
+of two can be used to determine when no more jobs are waitable when using
+``flux job wait`` in a loop.
 
 ``flux job wait --all`` loops through waitable jobs as they complete, printing
 their jobids.  If all jobs are successful, it exits with a code of zero.  If
```
It might be worth saying "loops through all the waitable jobs" here (adding "all the"), just to be clear.

chu11 replied:
i added an extra commit on top to just add this little clarification

garlick reviewed (approved):
Man page could maybe use some slight changes but otherwise looks good to me,  thanks!

chu11 commented:
@garlick thanks for the review.  added an extra commit on top of your clarification addition and fixed up that minor bit of text consistency.  Setting MWP

chu11 commented:
@Mergifyio rebase

chu11 commented:
@Mergifyio refresh

chu11 commented:
@mergifyio rebase

chu11 commented:
@mergifyio update

chu11 commented:
@Mergifyio requeue

chu11 commented:
@Mergifyio rebase
